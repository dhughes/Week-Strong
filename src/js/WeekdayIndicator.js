import React from 'react';
import { Hbox } from './Box';
import Day from './Day';

const WeekdayIndicator = props => {
  let days = [];
  for (let x = 0; x <= 6; x++) {
    days.push(
      <Day
        key={x}
        isSelected={props.selected.includes(x)}
        wasMissed={props.missed.includes(x)}
        dayOfWeek={x}
        today={props.today}
        onClick={props.onDayClick}
      />
    );
  }
  return (
    <Hbox>
      {days}
    </Hbox>
  );
};

WeekdayIndicator.defaultProps = {
  selected: [],
  missed: [],
  onDayClick: () => {}
};

export default WeekdayIndicator;
