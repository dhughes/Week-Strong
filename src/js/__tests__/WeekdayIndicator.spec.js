import React from 'react';
import { shallow } from 'enzyme';
import WeekdayIndicator from '../WeekdayIndicator';

test('WeekdayIndicator renders', () => {
  const component = shallow(<WeekdayIndicator />);
  expect(component).toMatchSnapshot();
});

test('WeekdayIndicator should show seven boxes', () => {
  const component = shallow(<WeekdayIndicator />);
  expect(component.find('Day').length).toEqual(7);
});
