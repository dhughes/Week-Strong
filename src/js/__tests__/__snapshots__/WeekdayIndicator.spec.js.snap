// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekdayIndicator renders 1`] = `
<styled.div>
  <Day
    dayOfWeek={0}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={1}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={2}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={3}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={4}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={5}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
  <Day
    dayOfWeek={6}
    isSelected={false}
    onClick={[Function]}
    wasMissed={false}
  />
</styled.div>
`;
